vispy/scene/widgets/widget.py

```diff
@@ -44,9 +45,15 @@ class Widget(Compound):
     def __init__(self, pos=(0, 0), size=(10, 10), border_color=None,
                  border_width=1, bgcolor=None, padding=0, margin=0, **kwargs):
         # For drawing border.
-        # A mesh is required because GL lines cannot be drawn with predictable
-        # shape across all platforms.
-        self._mesh = MeshVisual(color=border_color, mode='triangles')
+        self._border = _BorderVisual(pos=pos,
```

Comment:
This change causes a possible performance regression. We can benchmark this to see if it makes a real difference using `examples/basics/scene/grid_large.py`, however that example appears to be broken at the moment..